packages/unpublished-adapter-encapsulation-test-app/tests/integration/reload-test.js

```diff
@@ -0,0 +1,173 @@
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import EmberObject from '@ember/object';
+import Store from 'adapter-encapsulation-test-app/services/store';
+import Model, { attr } from '@ember-data/model';
+import JSONAPISerializer from '@ember-data/serializer/json-api';
+import Transform from '@ember-data/serializer/transform';
+import { resolve } from 'rsvp';
+
+class Person extends Model {
+  @attr
+  firstName;
+
+  @attr
+  lastName;
+}
+
+function setupReloadTest(options) {
+  class TestMinimumAdapter extends EmberObject {
+    shouldReloadAllCalled = 0;
+    shouldReloadRecordCalled = 0;
+    shouldBackgroundReloadAllCalled = 0;
+    shouldBackgroundReloadRecordCalled = 0;
+
+    requestsMade = 0;
+
+    constructor() {
+      super(...arguments);
+
+      if (options.shouldReloadAll !== undefined) {
+        this.shouldReloadAll = function() {
+          this.shouldReloadAllCalled++;
+          return options.shouldReloadAll;
+        };
+      }
+
+      if (options.shouldReloadRecord !== undefined) {
+        this.shouldReloadRecord = function() {
+          this.shouldReloadRecordCalled++;
+          return options.shouldReloadRecord;
+        };
+      }
+      if (options.shouldBackgroundReloadAll !== undefined) {
+        this.shouldBackgroundReloadAll = function() {
+          this.shouldBackgroundReloadAllCalled++;
+          return options.shouldBackgroundReloadAll;
+        };
+      }
+
+      if (options.shouldBackgroundReloadRecord !== undefined) {
+        this.shouldBackgroundReloadRecord = function() {
+          this.shouldBackgroundReloadRecordCalled++;
+          return options.shouldBackgroundReloadRecord;
+        };
+      }
+    }
+
+    findAll() {
+      this.requestsMade++;
+      return resolve(options.resolveFindAllWith || { data: [] });
+    }
+
+    findRecord() {
+      this.requestsMade++;
+      return resolve(options.resolveFindRecordWith || { data: null });
+    }
+  }
+  this.owner.register('adapter:application', TestMinimumAdapter);
+
+  this.store = this.owner.lookup('service:store');
+  this.adapter = this.owner.lookup('adapter:application');
+}
+
+module('integration/reload - Reloading Tests', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    // Needed to avoid deprecation warning even though not using any transforms.
+    this.owner.register('transform:date', class DateTransform extends Transform {});
+    this.owner.register('transform:number', class NumberTransform extends Transform {});
+    this.owner.register('transform:boolean', class BooleanTransform extends Transform {});
+    this.owner.register('transform:string', class StringTransform extends Transform {});
+
+    this.owner.register('service:store', Store);
+    this.owner.register('serializer:application', JSONAPISerializer);
```

Comment:
```ts
class Serializer extends EmberObject {
  normalizeResponse(_, __, data) { return data; }
}
```
^ this should be all this test needs

Reply:
pushPayload is also required I think. I'll try it out.

Reply:
pushPayload shouldn't be required as we aren't using it in the tests at all

Reply:
Seems to work. I thought it was needed because we call store.push.

Reply:
Nope :) only store.pushPayload uses serializer.pushPayload, it is called at no other time.